qpc/utils.py

```diff
@@ -124,6 +130,49 @@ def read_require_auth():
     return config.get(CONFIG_REQUIRE_TOKEN)
 
 
+def read_insights_config():
+    """Retrieve insights configuration.
+
+    :returns: The validated dictionary with configuration
+    """
+    if not os.path.exists(INSIGHTS_CONFIG):
+        return DEFAULT_INSIGHTS_CONFIG
+
+    with open(INSIGHTS_CONFIG) as insights_config_file:
+        try:
+            config = json.load(insights_config_file)
+        except exception_class:
+            return DEFAULT_INSIGHTS_CONFIG
+
+        host = config.get(CONFIG_HOST_KEY)
+        port = config.get(CONFIG_PORT_KEY)
+        use_http = config.get(CONFIG_USE_HTTP, False)
```

Comment:
If any of these keys don't exist (and we make impossible to save them as empty strings (`''`), we could simply run

```

        host = config.get(CONFIG_HOST_KEY, DEFAULT_INSIGHTS_CONFIG['host'])
        port = config.get(CONFIG_PORT_KEY, DEFAULT_INSIGHTS_CONFIG['port'])
        use_http = config.get(CONFIG_USE_HTTP, DEFAULT_INSIGHTS_CONFIG['use_http'])
```
We could also make a copy of default config passing the keys read from configuration (if we make sure we are not saving None or "empty" values

```
insights_config = dict(DEFAULT_INSIGHTS_CONFIG, **config)

```

Reply:
I went with the suggested approach. I added a function to validate the host and I update a copy of the default insights config dict with the received values from cli.